src/combinatorics/quiver/single_obj.lean

```diff
@@ -0,0 +1,111 @@
+/-
+Copyright (c) 2022 Antoine Labelle. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Labelle
+-/
+import combinatorics.quiver.connected_component
+
+/-!
+# Single-object quiver
+
+Single object quiver with a given edge type.
+
+## Main definitions
+
+Given a type `α`, `single_obj α` is `unit` type, whose single object is called `star α`, with
+`quiver` structure such that `star α ⟶ star α` is the type `α`.
+An element `x : α` can be reinterpreted as an element of `star α ⟶ star α` using
+`to_hom`. More generally, a list of elements of `a` can be reinterpreted as a path from `star α` to
+itself using `path_equiv_list`.
+-/
+
+namespace quiver
+
+/-- Type tag on `unit` used to define single-object quivers. -/
+@[nolint unused_arguments]
+def single_obj (α : Type*) : Type := unit
+
+namespace single_obj
+
+variables (α β γ : Type*)
+
+instance : quiver (single_obj α) := ⟨λ _ _, α⟩
+
+/-- The single object in `single_obj α`. -/
+def star : single_obj α := unit.star
+
+instance : inhabited (single_obj α) := ⟨star α⟩
+
+variables {α β γ}
+
+/-- The `has_reverse` structure on `single_obj α` given a fonction on `α`. -/
+def has_reverse (rev : α → α) : has_reverse (single_obj α) := ⟨λ _ _, rev⟩
```

Comment:
I think it still makes sense to have it as a definition, in any case, it can't hurt. We can always make them instances in specific cases where we have specific `rev` given by using these definitions.